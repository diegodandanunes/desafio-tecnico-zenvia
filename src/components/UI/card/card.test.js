import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import { Card } from './';
import fakeData from '../../../__mocks__/character-details';
const history = createMemoryHistory();

afterEach(cleanup);

it("renders", () => {
    const { asFragment } = render(<Router history={history}><Card info={fakeData.data[0]} /></Router>);
    expect(asFragment()).toMatchSnapshot();
});

it('correctly sets up card image tag', () => {
    const { getByTestId } = render(<Router history={history}><Card info={fakeData.data[0]} /></Router>);
    expect(getByTestId('card-image').src).toMatch(fakeData.data[0].img);
    expect(getByTestId('card-image').alt).toMatch(fakeData.data[0].name);
});

it('correctly links character details URL', () => {
    const { getByTestId } = render(<Router history={history}><Card info={fakeData.data[0]} /></Router>);
    expect(getByTestId('card-link').href).toMatch(`character/${fakeData.data[0].char_id}`);
});
