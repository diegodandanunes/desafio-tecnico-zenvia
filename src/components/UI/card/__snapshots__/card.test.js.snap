// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm fhxsoW"
  >
    <a
      data-testid="card-link"
      href="/character/1"
    >
      <div
        class="card-overlay"
      />
      <img
        alt="Neo"
        data-testid="card-image"
        src="https://observatoriodocinema.uol.com.br/wp-content/uploads/2020/04/Keanu-Reeves-1.jpg"
      />
    </a>
  </div>
</DocumentFragment>
`;
